azure-pipelines.yml

```diff
@@ -609,9 +609,18 @@ jobs:
       displayName: 'Performing the Tests'
 
     - bash: |
-        cd bin
+        git_hash=$(git rev-parse --verify HEAD) # of the last commit
+        date=$(date '+%Y.%m.%d')                # date in "YYYY.mm.dd" form
 
+        cd bin
         version=$(grep -i -e "project_version:*" CMakeCache.txt | cut -c24-)
+        ### In nonRelease builds
+        ### append date and git commit hash to the version. 
+        ### This way devs won't have to change the patch number on PRs
+        if [[ $version == *"nonRelease"* ]]; then
+            version=${version}_${date}_${git_hash}
+        fi
```

Comment:
What about this? @sarthakpati @ashishsingh18

Reply:
This checks if nonRelease is in the version and then does the date & hash things.

So basically the workflow should be that nonRelease is always in the cmake version, but when you do a PR for a stable release you remove the nonRelease part and azure builds and releases it normally